logserver/src/main/java/alluxio/logserver/AlluxioLogServerProcess.java

```diff
@@ -0,0 +1,307 @@
+        return;
+      }
+      int failureCount = 0;
+      while (!mStopped) {
+        try {
+          Socket client = mServerSocket.accept();
+          InetAddress inetAddress = client.getInetAddress();
+          // Currently loggerRepository is always null because we are not using the map.
+          LoggerRepository loggerRepository = mInetAddressHashMap.get(inetAddress);
+          if (loggerRepository == null) {
+            try {
+              loggerRepository = configureHierarchy(inetAddress, mLoggerType);
+              mInetAddressHashMap.put(inetAddress, loggerRepository);
+            } catch (URISyntaxException e) {
+              LOG.warn("URI syntax error.");
+              continue;
+            }
+          }
+          AlluxioLog4jSocketNode clientSocketNode =
+              new AlluxioLog4jSocketNode(client, loggerRepository);
+          int retryCount = 0;
+          long remainTimeInMillis = mRequestTimeoutInMillis;
+          while (true) {
+            try {
+              mThreadPool.execute(clientSocketNode);
+              break;
+            } catch (Throwable t) {
+              if (t instanceof RejectedExecutionException) {
+                retryCount++;
+                try {
+                  if (remainTimeInMillis > 0) {
+                    long sleepTimeInMillis = ((long) (mRandom.nextDouble()
+                        * (1L << Math.min(retryCount, 20)))) * mBackoffSlotInMillis;
+                    sleepTimeInMillis = Math.min(sleepTimeInMillis, remainTimeInMillis);
+                    TimeUnit.MILLISECONDS.sleep(sleepTimeInMillis);
+                    remainTimeInMillis -= sleepTimeInMillis;
+                  } else {
+                    client.close();
+                    LOG.warn("Connection has been rejected by ExecutorService "
+                        + retryCount + " times till timedout, reason: " + t);
+                    break;
+                  }
+                } catch (InterruptedException e) {
+                  LOG.warn("Interrupted while waiting to place client on executor queue.");
+                  Thread.currentThread().interrupt();
+                  break;
+                }
+              } else if (t instanceof  Error) {
+                LOG.error("ExecutorService threw error: " + t, t);
+                throw (Error) t;
+              } else {
+                LOG.warn("ExecutorService threw error: " + t, t);
+                break;
+              }
+              // Could not create a thread to serve a client, ignore this client.
+              LOG.warn("Failed to connect with client: {}.", inetAddress.toString());
+              continue;
+            }
+          }
+        } catch (IOException e) {
+          if (!mStopped) {
+            ++failureCount;
+            LOG.warn("Socket transport error occurred during accepting message.", e);
+          }
+          break;
+        }
+      }
+
+      mThreadPool.shutdown();
```

Comment:
Why does this class shutdown the thread pool? It is owned by the other class, and in this version, there will be two instances of `LogReceiver`, right?

Reply:
Done.